contracts/modules/Offers/Omnibus/OffersOmnibus.sol

```diff
@@ -0,0 +1,350 @@
+        }
+
+        if (_findersFeeBps > 0) {
+            require(_findersFeeBps <= 10000, "createOffer finders fee bps must be less than or equal to 10000");
+            require(_findersFeeBps + _listingFee.listingFeeBps <= 10000, "listingFee and findersFee must be less than or equal to 10000");
+            _setFindersFee(offer, _findersFeeBps);
+        }
+
+        if (_expiry > 0) {
+            require(_expiry > block.timestamp, "Expiry must be in the future");
+            _setExpiry(offer, _expiry);
+        }
+
+        emit OfferCreated(_tokenContract, _tokenId, offerCount, _getFullOffer(offer));
+        return offerCount;
+    }
+
+    function _validateAndPrepareNewOffer(
+        address _oldCurrency,
+        address _newCurrency,
+        uint256 _oldAmount,
+        uint256 _newAmount
+    ) internal {
+        require(_newAmount > 0, "NO_ZERO_OFFERS");
+        require(_newAmount != _oldAmount || _newCurrency != _oldCurrency, "SAME_OFFER");
+
+        IERC20 token;
+
+        if (_newCurrency == address(0)) {
+            if (_oldCurrency == address(0)) {
+                if (_newAmount > _oldAmount) {
+                    require(msg.value == _newAmount - _oldAmount, "INVALID_MSG_VALUE");
+                } else {
+                    require(msg.value == 0, "INVALID_MSG_VALUE");
+                }
+            } else {
+                require(msg.value == _newAmount, "INVALID_MSG_VALUE");
+            }
+            token = IERC20(address(weth));
+        } else {
+            require(msg.value == 0, "INVALID_MSG_VALUE");
+            token = IERC20(_newCurrency);
+        }
+
+        if (msg.value > 0) {
+            weth.deposit{value: msg.value}();
+            weth.transferFrom(address(this), msg.sender, msg.value);
+        }
+
+        require(token.balanceOf(msg.sender) >= _newAmount, "INSUFFICIENT_BALANCE");
+        require(token.allowance(msg.sender, address(erc20TransferHelper)) >= _newAmount, "INSUFFICIENT_ALLOWANCE");
+    }
+
+    /// @notice Updates the price of the given offer
+    /// @param _tokenContract The address of the offer ERC-721 token
+    /// @param _tokenId The ID of the offer ERC-721 token
+    /// @param _offerId The ID of the offer
+    /// @param _offerCurrency The address of the ERC-20 token offered
+    /// @param _offerAmount The new amount offered
+    function setOfferAmount(
+        address _tokenContract,
+        uint256 _tokenId,
+        uint256 _offerId,
+        address _offerCurrency,
+        uint256 _offerAmount
+    ) external payable nonReentrant {
+        StoredOffer storage offer = offers[_tokenContract][_tokenId][_offerId];
+
+        require(offer.maker == msg.sender, "CALLER_NOT_MAKER");
+
+        _validateAndPrepareNewOffer(_getERC20CurrencyWithFallback(offer), _offerCurrency, offer.amount, _offerAmount);
+
+        require(erc721TransferHelper.isModuleApproved(msg.sender), "MODULE_NOT_APPROVED");
```

Comment:
If this is an offer, I don't know if we need to be checking for module approval.